### PR TITLE
Add tutorial for no-code dataset upload

### PR DESCRIPTION
**Pull request by stevhliu**
This PR is for a tutorial for uploading a dataset to the Hub. It relies on the Hub UI elements to upload a dataset, introduces the online tagging tool for creating tags, and the Dataset card template to get a head start on filling it out. The addition of this tutorial should make it easier for beginners to upload a dataset without accessing the terminal or knowing Git.

### CODE REVIEW

**Comment by lhoestq** [comment]
Cool, love it ! :)

Feel free to add a paragraph saying how to load the dataset:
```python
from datasets import load_dataset

dataset = load_dataset("stevhliu/demo")

# or to separate each csv file into several splits
data_files = {"train": "train.csv", "test": "test.csv"}
dataset = load_dataset("stevhliu/demo", data_files=data_files)
print(dataset["train"][0])
```

**Comment by lhoestq** [comment]
Perfect, feel free to mark this PR ready for review :)

cc @albertvillanova do you have any comment ? You can check the tutorial here:
https://47389-250213286-gh.circle-artifacts.com/0/docs/_build/html/no_code_upload.html

Maybe we can just add a list of supported file types:
- csv
- json
- json lines
- text
- parquet

**Review by lhoestq — APPROVED** [review]

**Comment by lhoestq** [comment]
I just added a mention of the login for private datasets. Don't hesitate to edit or comment.

Otherwise I think it's all good, feel free to merge it @stevhliu  if you don't have other changes to make :)

File: docs/source/upload_dataset.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+3. Once you have uploaded your dataset files, they are now stored in your dataset repository.
+
+.. image:: /imgs/files_stored.png
+   :align: center
+
+Create a Dataset card
+---------------------
+
+The last step is to create a Dataset card. The Dataset card is very important for helping users understand how to use your dataset responsibly.
```
Comment (albertvillanova) [inline docs/source/upload_dataset.rst:40]:
```suggestion
The last step is to create a Dataset card. The Dataset card is very important for helping users find your dataset and understand how to use it responsibly.
```

**Review by albertvillanova — APPROVED** [review]
Awesome! This tutorial is very useful and simple. Thanks! 🤗 

Only a minor comment below.

**Pull request merged**